virtcontainers/kata_agent.go

```diff
@@ -1084,7 +1083,12 @@ func (k *kataAgent) buildContainerRootfs(sandbox *Sandbox, c *Container, rootPat
 			rootfs.Source = blockDrive.VirtPath
 		} else if sandbox.config.HypervisorConfig.BlockDeviceDriver == config.VirtioBlock {
 			rootfs.Driver = kataBlkDevType
-			rootfs.Source = blockDrive.PCIAddr
+			if blockDrive.PCIAddr == "" {
+				rootfs.Source = blockDrive.VirtPath
```

Comment:
Is there a reason you want to provide the virtpath instead of PCIAddr here?

Reply:
This change corresponds to https://github.com/kata-containers/agent/pull/574. In the runtime, since the virpath is known (as the container rootfs in not hot-plugged but a dummy one is created and later updated), it is better to provide this information to the agent directly, rather than providing PCIAddr and then agent finding out the virtpath.

Reply:
Fair enough.